src/library/Truncate/Truncate.js

```diff
@@ -0,0 +1,106 @@
+/* @flow */
+import React, { Component } from 'react';
+import { canUseDOM } from 'exenv';
+import debounce from 'lodash.debounce';
+import { ellipsis } from 'polished';
+import { createStyledComponent } from '../styles';
+import _Tooltip from '../Tooltip';
+
+type Props = {
+  /** Content of Truncate */
+  children?: React$Node,
+  /** Maximum width of the content */
+  maxWidth?: number | string
+};
+
+type State = {
+  tooltip: boolean
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  // Truncate_borderColor_focus: baseTheme.borderColor_theme_focus,
+
+  ...baseTheme
+});
+
+const styles = {
+  root: ({ maxWidth, theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      pointerEvents: 'all', // Necessary because of Button Inner's pointerEvents: none
+      ...ellipsis(maxWidth),
+
+      '&:focus': {
+        outline: `1px solid ${theme.color_theme}`,
+        outlineOffset: -1
+      }
+    };
+  },
+  tooltip: {
+    // This style is only necessary because we're using Truncate as a Button child
+    whiteSpace: 'normal'
+  }
+};
+
+const Root = createStyledComponent('span', styles.root, {
+  displayName: 'Truncate'
+});
+const Tooltip = createStyledComponent(_Tooltip, styles.tooltip);
+
+export default class Truncate extends Component<Props, State> {
+  static defaultProps = {
+    maxWidth: '100%'
+  };
+
+  state = {
+    tooltip: false
+  };
+
+  root: HTMLElement;
+
+  setRootRef = (node: HTMLElement) => {
+    this.root = node;
+  };
+
+  addTooltip = () => {
+    const rootNode = this.root;
+
+    if (canUseDOM && rootNode) {
+      const offsetWidth = rootNode.offsetWidth + 1; // `+ 1` necessary for Edge & IE11
+      const tooltip = Boolean(rootNode.scrollWidth > offsetWidth);
```

Comment:
Can we rename the `tooltip` const to `showTooltip` or similar?  Along with the aforementioned state key change to `hasTooltip`, it would make the later comparisons more intuitive.